lib/bridge/MatrixHandler.js

```diff
@@ -770,6 +769,83 @@ MatrixHandler.prototype._onMessage = Promise.coroutine(function*(req, event) {
     yield Promise.all(promises);
 });
 
+MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
+    function*(req, ircRoom, ircClient, ircAction, event) {
+
+    // Send the action as is if it is not a text message 
+    //  Also, check for the existance of the getSplitMessages method.
+    if (event.content.msgtype !== "m.text" ||
+        !(ircClient.unsafeClient && ircClient.unsafeClient.getSplitMessages)) {
+        yield this.ircBridge.sendIrcAction(ircRoom, ircClient, ircAction);
+        return;
+    }
+
+    let text = event.content.body;
+
+    // Generate an array of individual messages that would be sent
+    let potentialMessages = ircClient.unsafeClient.getSplitMessages(ircRoom.channel, text);
+    let lineLimit = ircRoom.server.getLineLimit();
+
+    if (potentialMessages.length <= lineLimit) {
+        yield this.ircBridge.sendIrcAction(ircRoom, ircClient, ircAction);
+        return;
+    }
+
+    // Message body too long, upload to HS instead
+    try {
+        // Use the current time as the name of the file
+        //  strip off day of week and timezone
+        //  replace whitespace with underscores
+        // result e.g : Aug_01_2016_13:41:44.txt
+        let fileName = new Date().toString()
+            .match(/.*?\s(.*\:[0-9]*\:[0-9]*).*/)[1]
+            .replace(/\s/g,'_') + ".txt";
+
+        // Try to upload as a file and get URI
+        //  (this could fail, see the catch statement)
+        let response = yield this.ircBridge.sendMatrixPlaintextFile(fileName, text);
```

Comment:
Naming: `send` makes me think it is sending the message but that's not what it's really doing, it's *uploading* the text. I would prefer if this were called `uploadTextFile` or equiv (we don't have anywhere else that we can upload to, so the "matrix" bit is somewhat redundant).